### ISSUE TITLE
problem with underscore in external database name?

### ISSUE THREAD

**Issue by slambrechts** [jtamames/SqueezeMeta#404]
Hi,

I used a few external databases, and when I run `sqm2tables.py` it gives the following error:

```
(SqueezeMeta) slambrecht@WE11sv03:~/MICROBIAN_metagenomics/SQM/microbian43$ sqm2tables.py /home/slambrecht/MICROBIAN_metagenomics/SQM/microbian43/microbian43 /home/slambrecht/MICROBIAN_metagenomics/SQM/microbian43/microbian43/results/tables --force-overwrite
Traceback (most recent call last):
  File "/home/slambrecht/miniconda2/envs/SqueezeMeta/bin/sqm2tables.py", line 218, in <module>
    main(parse_args())
  File "/home/slambrecht/miniconda2/envs/SqueezeMeta/bin/sqm2tables.py", line 70, in main
    args.trusted_functions, args.ignore_unclassified, customMethods, data_dir)
  File "/home/slambrecht/miniconda2/envs/SqueezeMeta/SqueezeMeta/utils/../lib/utils.py", line 218, in parse_orf_table
    ID = line[idx[method]].replace('*', '')
KeyError: 'hydrogenase_subgroups'

```
Since I used SQM before with external databases without issues, is it possible this error is due to the underscore in the database name `hydrogenase_subgroups`? The reason I ask is because the external database that worked before was named `custom`, but now I named this new database `hydrogenase_subgroups` and the error occurs.

Or is this due to something completely different?

**Comment by fpusan** [comment]
Hi!
Can you send me the first 10 lines of the 13.*.oftable file for that project?
Merry christmas

**Comment by slambrechts** [comment]
Hi!

I think I can answer this myself now:

So I was using multiple external databases, but I needed a different `$minidenfun4` for some of them. So each time I changed `$minidenfun4` in the `parameters.pl` file, changed the` mydb.list` file to only that specific external database that I need to run at that `$minidenfun4`, and ran `04.rundiamond.pl <project name> 1` each time following the procedure suggested in  #https://github.com/jtamames/SqueezeMeta/issues/400. But in the end when rerunning scripts 13, 22, and `sqm2tables.py`, I forgot to put all the external databases I used back into the `mydb.list` file, which is what was causing this error.

Also, it seems the order in which you put the external databases back into the mydb.list file at the end matters? In any case, when looking at the resulting tables, some of them have the wrong name. For example when I want to check the results of the custom database with hydrogenase genes I used, the `microbian43.hydrogenase.copyNumber.tsv` table contains the results of another completely different external database I used. At the moment I haven't tested yet whether this is due to the order of the databases in the mydb.list file. Other then that, it seems to work fine.
